src/associatedLegendre.jl

```diff
@@ -0,0 +1,594 @@
+            #@test_approx_eq_eps a b 1e-14
+            end
+        end
+    
+    end
+
+end
+testSSLegendreP2()
+
+#checking SSLegendreP2 against Float64(SLegendreP2)
+#   for n< 17 the relative error is less than 3e-11
+# the error depends on the n amd m values and x values to some extent
+"""
+    function testSSLegendreP2F()
+    compare SSLegendreP2 to float64SLegendreP2
+"""
+function testSSLegendreP2F()
+    x=.6
+    rerr = 3e-11
+    for n=0:17
+        for m=0:n    #m=rand(0:n)
+            a=SSLegendreP2(n,m,x)
+            b=Float64(SLegendreP2(n,m,big(x)))
+            df=norm(a-b)/norm(b) 
+            if ( df > rerr)
+            println(n," ", m," ", a," ",b," ",df)
+            end
+        end
+    end
+end
+testSSLegendreP2F()
+
+
+"""
+SLegendreP3(n::Integer,m::Integer,z::Number)
+    associated Legendre function z>1 1st kind type 3
+    series expansion
+"""
+function SLegendreP3(n::Integer,m::Integer,z::Number)
+#function mylegenp3(n,m,z)#n,m integers 
+    v=(z*z - big(1.))^(-m*big(1)/2)
+  
+    v=v*factorial(big(n+m))/(((big(2))^n)* factorial(big(n)));
+    term =  big(0.);
+         #if iseven(n+m); MXP=Int(trunc((n+m)/2));end;
+         #if isodd(n+m);MXP=Int(trunc((n+m-1)/2));end;
+         #MXP=Int(floor((n+m)/2));
+    MXP=Int(trunc(((n+m)/2)))
+    
+    for p=0:MXP
+                                                          
+        term= term + 
+        ((-big(1.))^p)*binomial(big(2*(n-p)),big(n-m))*
+        (z^(big(1)*(n + m -2*p)))*
+        binomial(big(n),big(p)) 
+        
+    end
+    return term * v
+end   
+
+#print mylegenp3(2,1,2.)#agrees to 16 decimals
+#print mylegenp3(2,1,mpf(2))#agrees
+#print mylegenp3(2,1,mpf('2.'))#agrees 10.392304845413263761
+#println(mylegenp3(2,1,big(2.)))  AGREES with mpmath
+
+# I derived this finite sum using  8.6.6,8.6.18,8.2.5 (AS),
+#binomial expansion and differentiating
+
+
+function timeS2()
+    for i=1:10
+        SLegendreP2(100,50,big(.6))
+    end
+end   
+TS2= @elapsed(timeS2())
+println("time SLegendreP2(100,50,big(.6))  $TS2")
+
+function timeP2()
+    for i=1:10
+        SSLegendreP2(100,50,big(.6))
+    end
+end   
+TP2= @elapsed(timeP2())
+println("time SSLegendreP2(100,50,big(.6))  $TP2")
+#time SLegendreP2(100,50,big(.6))  0.0085234
+#time SSLegendreP2(100,50,big(.6))  0.004890456
+# the series is double speed when each term is a multiple of previous term
+
+
+
+"""
+    SSLegendreP3(n::Integer,m::Integer,x::Number)
+    associated Legendre function 1st kind type 3 , x >1.
+    each term in sum is multiple of previous. 
+
+"""
+
+function SSLegendreP3(n::Integer,m::Integer,x::Number)#n,m integers 
+    
+   
+    M = (2*n -1) # M must be odd
+    # (2m-1)!!= (2m)!/( m! 2^m)   double factorial 
+    dblfac=1
+    for j=1:M
+        if  iseven(j)
+            continue
+        end    
+        dblfac=j*dblfac  
+    end
+    
+    MXP=Int(trunc(((n+m)/2)))
+    sum= (x^(n+m))*dblfac/(((x*x - one(x))^(m/2))*prod(1:n-m))
```

Comment:
This could have spurious overflow (the numerator and/or denominator could overflow even though the ratio is in a representable range)...